### DISCUSSION TITLE
cp_model_lns selecters and accepters

### DISCUSSION THREAD

**Discussion by C8504** [google/or-tools#4076] [CP-SAT questions] [answered]
Inside cp_model_lns, I found it used UCB algorithm as generaters select criteria, while I didn't see which kind accept criterias when a lns subsolver found a new solution. For example, if the new solution is better, we will accept only or we also accept worse solution to avoid local optimal? Could you please advice where I could find details?
Thanks a lot.

**Comment by lperron** [comment]
There is a solution pool that contains sub-optimal solutions.
LNS selects solutions from that pool.

**Reply by C8504** [reply]
Thanks lperron, while how to select so many solutions, always select better?

**Comment by lperron** [comment, marked as answer]
so, pool size is 3 by default ( I believe, there is a parameter). And it
picks randomly from that pool.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00

**Reply by C8504** [reply]
ok, thanks
